File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/data/DataFloat32.java
package tasks.data;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Float32;

@JavaTaskInfo
public class DataFloat32 extends JavaTask {

	@InputPortInfo(name = "In", dataType = Float32.class)
	public InputPort in;
	@OutputPortInfo(name = "Out", dataType = Float32.class)
	public OutputPort out;

	@Override
	public void execute() {
		Float32 inFloat32 = in.getData(Float32.class);
		out.putData(new Float32(inFloat32.getValue()*2));
	}

}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/data/DataFloat32.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Float32 import Float32

@PythonTaskInfo
class DataFloat32(PythonTask):

    def __init_ports__(self):
        self.in1 = InputPort(name="in1", data_type=Float32)
        self.in2 = InputPort(name="in2", data_type=Float32)
        self.out = OutputPort(name="out", data_type=Float32)

    def execute(self, grp):
        in1Data = self.in1.get_data(Float32)
        in2Data = self.in2.get_data(Float32)
        self.out.put_data(Float32(in1Data.values[0] + in2Data.values[0]))
